### ISSUE TITLE
{"message":"The following changes cannot be made in additive-only schema mode:\n- Property 'todo.items' has been changed from 'set<mixed>' to 'set<mixed>'.","errorCode":1}

### ISSUE THREAD

**Issue by genielabel** [realm/realm-js#5364]
### How frequently does the bug occur?

Always

### Description

"react": "18.2.0",
"react-native": "0.71.1",

I try to run the realm with the latest version but I get an error at the beginning of the synchronisation, and yet my schemas are impacable and identical

### Stacktrace & log output

```shell
{"message":"The following changes cannot be made in additive-only schema mode:\n-
 Property 'todo.items' has been changed from 'set<mixed>' to 'set<mixed>'.","errorCode":1}
```

### Can you reproduce the bug?

No

### Version

11.4.0

### What services are you using?

Atlas Device Sync

### Are you using encryption?

No

### Platform OS and version(s)

Android 10

### Build environment

Which debugger for React Native: ..

**Comment by kneth** [comment]
@genielabel Thank you for reporting. I will say that the error message is a bit strange as both types are the same.

* It is possible for you to share the full schema - both in your app and server-side? 
* Are you using developer mode (server-side)?
* Do you see the same error message if you remove and reinstall the app?

**Comment by genielabel** [comment]
@kneth 

server-side  { ..."items": {
      "bsonType": "array",
      "uniqueItems": true,
      "items": {
        "bsonType": "mixed"
      }
    ...}

Mobile Sdk  : 
{...
    items: '{}<>',
..
}

**Comment by kneth** [comment]
@genielabel Our schema parser is buggy, and we have rewritten it for the upcoming version 12 (see https://github.com/realm/realm-js/discussions/5416 for many more details). For now, I will suggest that you write the schema in the explicit/canonical form: `items: { objectType: "mixed", type: "list" }`

**Comment by kneth** [comment]
I am closing the issue as we have released our first alpha of v12.0.0
